File: go/dsa/algorithms/sliding-window/max-sum.go
func findMaxSumSubArray(arr []int, k int) int {
    left := 0
    windowSum := 0
    maxSum := -1 << 31

    for right := 0; right < len(arr); right++ {
        // 1️⃣ Expand
        windowSum += arr[right]

        // 2️⃣ When window size reaches k
        if right - left + 1 == k {
            // 3️⃣ Record / Evaluate
            if windowSum > maxSum {
                maxSum = windowSum
            }

            // 4️⃣ Slide window
            windowSum -= arr[left]
            left++
        }
    }
    return maxSum
}

File: python/dsa/algorithms/sliding-window/max-sum.py
def find_max_sum_subarray(arr, k):
    left = 0
    window_sum = 0
    max_sum = float('-inf')

    for right in range(len(arr)):
        # 1️⃣ Expand
        window_sum += arr[right]

        # 2️⃣ When window size reaches k
        if right - left + 1 == k:
            # 3️⃣ Record / Evaluate
            max_sum = max(max_sum, window_sum)

            # 4️⃣ Slide window
            window_sum -= arr[left]
            left += 1

    return max_sum